src/compiler/crystal/macros/methods.cr

```diff
@@ -2394,6 +2394,16 @@ private def interpret_array_or_tuple_method(object, klass, method, args, named_a
       end
       klass.new(object.elements + other_elements)
     end
+  when "-"
+    interpret_check_args(node: object) do |arg|
+      case arg
+      when Crystal::ArrayLiteral
```

Comment:
```suggestion
      when Crystal::TupleLiteral
        other_elements = arg.elements
      when Crystal::ArrayLiteral
```

This would make the implementation exhibit the same behaviour as `ArrayLiteral#+` and `TupleLiteral#+`, a  behaviour that does not appear to be spec'd